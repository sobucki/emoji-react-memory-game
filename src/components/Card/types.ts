export type CardProps = {
  value: string;
};
