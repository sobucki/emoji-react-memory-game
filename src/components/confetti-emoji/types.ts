export type ConfettiEmojiProps = {
  emojis: string[];
};
