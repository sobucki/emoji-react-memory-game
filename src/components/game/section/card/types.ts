import { HTMLAttributes } from "react";

export interface CardProps extends HTMLAttributes<HTMLButtonElement> {
  info: CardInfo;
  // onClick: (card: CardInfo, matchCard: (value: boolean) => void) => void;
}

export type CardData = Pick<CardInfo, "id" | "value">;

export type CardInfo = {
  id: string;
  value: string;
  status: CardStatus;
};

export type CardStatus = "hide" | "flipped" | "matched";
