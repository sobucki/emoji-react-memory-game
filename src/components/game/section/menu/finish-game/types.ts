export type MenuProps = {
  isOpen: boolean;
  moves: number;
  seconds: number;
  onStart: () => void;
};
