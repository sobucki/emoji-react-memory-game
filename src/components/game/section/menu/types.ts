export type MenuProps = {
  isOpen: boolean;
  onClose: () => void;
};
