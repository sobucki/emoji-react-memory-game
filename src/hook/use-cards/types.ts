export type Card = {
  id: string;
  value: string;
  isMatched: string;
  state: "flipped" | "hide";
};

export type ShuffleProps = {
  level: GameLevel;
  category: GameCategory;
};

export type GameCategory = "animals";

export type GameLevel =
  | "very_easy"
  | "easy"
  | "normal"
  | "hard"
  | "very_hard"
  | "insane";
