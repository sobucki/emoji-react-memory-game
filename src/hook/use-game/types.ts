export type useGameProps = {
  optionCards: string[];
};

export type StartGameProps = {
  category: string;
  level: string;
  optionsCards: string[];
};
