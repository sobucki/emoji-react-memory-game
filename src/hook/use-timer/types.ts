export type UseTimerProps = {
  initialValue?: number;
};
