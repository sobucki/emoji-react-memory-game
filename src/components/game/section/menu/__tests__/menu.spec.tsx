import { render, screen } from "@testing-library/react";
import Menu from "..";
import { MenuProps } from "../types";

const defaultProps: MenuProps = { isOpen: true, onClose: vi.fn() };

const renderMenu = (props: MenuProps = defaultProps) => {
  return render(<Menu {...props} />);
};

describe("menu", () => {
  it("should render correctly", () => {
    const { asFragment } = renderMenu();
    expect(
      screen.getByRole("heading", { name: /start a new game/i })
    ).toBeInTheDocument();

    expect(
      screen.getByText(/configure how you want to play/i)
    ).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it("should not open menu when pass isOpen false", () => {
    renderMenu({ ...defaultProps, isOpen: false });

    expect(
      screen.queryByRole("heading", { name: /start a new game/i })
    ).not.toBeInTheDocument();

    expect(
      screen.queryByText(/configure how you want to play/i)
    ).not.toBeInTheDocument();
  });

  it("should combobox have correctly options", () => {
    renderMenu();

    const category = screen.getByRole("combobox", { name: /category:/i });

    expect(category).toMatchInlineSnapshot(`
      <select
        id="category-select"
        name="categories"
      >
        <option
          value=""
        >
          --Please choose an category--
        </option>
        <option
          value="activity_sports"
        >
          Activity and Sports ⚽️🤸‍♂️
        </option>
        <option
          value="animals"
        >
          Animals & Nature 🐶 🐱
        </option>
        <option
          value="clothing_accessories"
        >
          Clothing and Accessories 🧦🕶
        </option>
        <option
          value="food_drink"
        >
          Food & Drink 🍕🍺
        </option>
        <option
          value="gestures_body"
        >
          Gestures and Body Parts 👋🦶
        </option>
        <option
          value="objects"
        >
          Objects 🔧💡
        </option>
        <option
          value="people_fantasy"
        >
          People and Fantasy 👨🧛‍♂️
        </option>
        <option
          value="smileys"
        >
          Smileys 😃😝
        </option>
        <option
          value="symbols"
        >
          Flags 🇧🇷🇺🇸
        </option>
        <option
          value="symbols"
        >
          Symbols 🆘🛂
        </option>
        <option
          value="travel_places"
        >
          Travel & Places 🚙 🌆
        </option>
      </select>
    `);

    const level = screen.getByRole("combobox", { name: /level:/i });

    expect(level).toMatchInlineSnapshot(`
      <select
        id="level-select"
        name="categories"
      >
        <option
          value=""
        >
          --Please choose an level--
        </option>
        <option
          value="very_easy"
        >
          🤩 Very easy (5 emojis)
        </option>
        <option
          value="easy"
        >
          🤗 Easy (8 emojis)
        </option>
        <option
          value="normal"
        >
          🙂 Normal (10 emojis)
        </option>
        <option
          value="hard"
        >
          😣 Hard (15 emojis)
        </option>
        <option
          value="very_hard"
        >
          😨 Very hard (20 emojis)
        </option>
        <option
          value="insane"
        >
          😱 Insane (all available)
        </option>
      </select>
    `);
  });
});
