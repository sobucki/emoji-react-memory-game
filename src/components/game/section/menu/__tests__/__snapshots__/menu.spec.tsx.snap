// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`menu > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-beySPh geyGaj"
  >
    <div
      class="sc-guDLey zOSuX"
    >
      <h2>
        Start a new game
      </h2>
      <fieldset>
        <legend>
          Configure how you want to play
        </legend>
        <div>
          <label
            for="category-select"
          >
            Category:
          </label>
          <select
            id="category-select"
            name="categories"
          >
            <option
              value=""
            >
              --Please choose an category--
            </option>
            <option
              value="activity_sports"
            >
              Activity and Sports ⚽️🤸‍♂️
            </option>
            <option
              value="animals"
            >
              Animals & Nature 🐶 🐱
            </option>
            <option
              value="clothing_accessories"
            >
              Clothing and Accessories 🧦🕶
            </option>
            <option
              value="food_drink"
            >
              Food & Drink 🍕🍺
            </option>
            <option
              value="gestures_body"
            >
              Gestures and Body Parts 👋🦶
            </option>
            <option
              value="objects"
            >
              Objects 🔧💡
            </option>
            <option
              value="people_fantasy"
            >
              People and Fantasy 👨🧛‍♂️
            </option>
            <option
              value="smileys"
            >
              Smileys 😃😝
            </option>
            <option
              value="symbols"
            >
              Flags 🇧🇷🇺🇸
            </option>
            <option
              value="symbols"
            >
              Symbols 🆘🛂
            </option>
            <option
              value="travel_places"
            >
              Travel & Places 🚙 🌆
            </option>
          </select>
          <label
            for="level-select"
          >
            Level:
          </label>
          <select
            id="level-select"
            name="categories"
          >
            <option
              value=""
            >
              --Please choose an level--
            </option>
            <option
              value="very_easy"
            >
              🤩 Very easy (5 emojis)
            </option>
            <option
              value="easy"
            >
              🤗 Easy (8 emojis)
            </option>
            <option
              value="normal"
            >
              🙂 Normal (10 emojis)
            </option>
            <option
              value="hard"
            >
              😣 Hard (15 emojis)
            </option>
            <option
              value="very_hard"
            >
              😨 Very hard (20 emojis)
            </option>
            <option
              value="insane"
            >
              😱 Insane (all available)
            </option>
          </select>
          <button>
            Start
          </button>
        </div>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
